community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -131,6 +132,7 @@ public final void close()
             if ( cursor.pagedFile != null )
             {
                 cursor.unpinCurrentPage();
+                cursor.pageCursorTracer.reportEvents();
```

Comment:
I'm not entirely convinced that this is the right place to put it. Cypher wants to get numbers for a whole Statement, in which case it makes sense to collect and report the data when a statement is closed. Linked cursors would presumably also be attached to the same thread-local tracer, in which case there is no point to do the reporting several times.